flow/probes/ebpf.go

```diff
@@ -80,298 +71,24 @@ type EBPFProbesHandler struct {
-				path, app := flow.LayersPath(p.Layers())
-				f.LayersPath += "/" + path
-				f.Application = app
-			}
-			*/
-		}
-	}
-
-	// ICMP
-	if layersInfo&uint8(C.ICMP_LAYER_INFO) > 0 {
-		kind := uint8(kernFlow.icmp_layer.kind)
-		code := uint8(kernFlow.icmp_layer.code)
-		id := uint16(kernFlow.icmp_layer.id)
-
-		f.ICMP = &flow.ICMPLayer{
-			Code: uint32(code),
-			ID:   uint32(id),
-		}
-		if f.Network != nil && f.Network.Protocol == flow.FlowProtocol_IPV4 {
-			f.ICMP.Type = flow.ICMPv4TypeToFlowICMPType(kind)
-		} else {
-			f.ICMP.Type = flow.ICMPv6TypeToFlowICMPType(kind)
-		}
-	}
-
-	appLayers := strings.Split(f.LayersPath, "/")
-	f.Application = appLayers[len(appLayers)-1]
-
-	f.Metric = &flow.FlowMetric{
-		ABBytes:   int64(kernFlow.metrics.ab_bytes),
-		ABPackets: int64(kernFlow.metrics.ab_packets),
-		BABytes:   int64(kernFlow.metrics.ba_bytes),
-		BAPackets: int64(kernFlow.metrics.ba_packets),
-		Start:     f.Start,
-		Last:      f.Last,
-	}
-
-	key := kernFlowKey(kernFlow)
-
-	f.UpdateUUID(key, flow.Opts{LayerKeyMode: flow.L3PreferedKeyMode})
-
-	fops = append(fops, &flow.Operation{
-		Type: flow.ReplaceOperation,
-		Flow: f,
-		Key:  key,
-	})
-	return fops
-}
-
-func (p *EBPFProbe) updateFlowOperation(ebpfFlow *EBPFFlow, kernFlow *C.struct_flow, startKTimeNs int64, start time.Time) *flow.Operation {
-	f := flow.NewFlow()
-	f.Last = common.UnixMillis(ebpfFlow.last)
-	layersInfo := uint8(kernFlow.layers_info)
-	if layersInfo&uint8(C.TRANSPORT_LAYER_INFO) > 0 {
-		protocol := uint8(kernFlow.transport_layer.protocol)
-		switch protocol {
-		case syscall.IPPROTO_TCP:
-			f.TCPMetric = &flow.TCPMetric{
-				ABSynStart: tcpFlagTime(kernFlow.transport_layer.ab_syn, startKTimeNs, start),
-				BASynStart: tcpFlagTime(kernFlow.transport_layer.ba_syn, startKTimeNs, start),
-				ABFinStart: tcpFlagTime(kernFlow.transport_layer.ab_fin, startKTimeNs, start),
-				BAFinStart: tcpFlagTime(kernFlow.transport_layer.ba_fin, startKTimeNs, start),
-				ABRstStart: tcpFlagTime(kernFlow.transport_layer.ab_rst, startKTimeNs, start),
-				BARstStart: tcpFlagTime(kernFlow.transport_layer.ba_rst, startKTimeNs, start),
-			}
-		}
-	}
-	f.Metric = &flow.FlowMetric{
-		ABBytes:   int64(kernFlow.metrics.ab_bytes),
-		ABPackets: int64(kernFlow.metrics.ab_packets),
-		BABytes:   int64(kernFlow.metrics.ba_bytes),
-		BAPackets: int64(kernFlow.metrics.ba_packets),
-		Start:     f.Start,
-		Last:      f.Last,
-	}
-
-	key := kernFlowKey(kernFlow)
-
-	return &flow.Operation{
-		Type: flow.UpdateOperation,
-		Flow: f,
-		Key:  key,
-	}
-}
-
 func (p *EBPFProbe) run() {
 	var info syscall.Sysinfo_t
 	syscall.Sysinfo(&info)
 
-	_, flowChanOperation := p.flowTable.Start()
+	_, flowChanOperation, _ := p.flowTable.Start()
 	defer p.flowTable.Stop()
 
 	var startKTimeNs int64
 	var start time.Time
 
-	ebpfFlows := make(map[C.__u64]*EBPFFlow)
 	updateTicker := time.NewTicker(ebpfUpdate)
 	defer updateTicker.Stop()
 
+	p.flowPoolSize = 2*cap(flowChanOperation) + 1
```

Comment:
p := 2*cap....
not need to have it as struct member

Reply:
+1 is redundant so I removed it;)
this is no longer a struct member as well.
As for your comment @nplanel -- I can indeed incorporate this change you are suggesting, but I still don't understand why it's needed. Eran also wrote a comment for u on this. Would be much appreciated if you could take a look, because both of us don't see why the current implementation is wrong.
+ adding a channel in the opposite direction is blocking as well and harms performance so would be smart to avoid this if the code is working as is, right?